### PR TITLE
Adds REST+Virtual Filesystem integration test case

### PR DESCRIPTION
**Pull request by hellofromtonya**
- Adds a `RESTVfsTestCase` which allows uses our virtual filesystem (vfs) + WordPress' REST API
- Moves common REST stuff into a trait
- Updates the `RESTfulTestCase` to use the trait

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@crystinutzaa I moved your REST+Vfs test case code over to this package, so that we can reuse it when needed (which I suspect will be needed beyond CPCSS).

File: Integration/RESTTrait.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	/**
+	 * Sets up the WP REST API Server.
+	 */
+	protected function setUpServer() {
+		/**
+		 * @var WP_REST_Server $wp_rest_server
+		 */
+		global $wp_rest_server;
+		$this->server = $wp_rest_server = new WP_REST_Server;
+		do_action( 'rest_api_init' );
```
Comment (Screenfeed) [inline Integration/RESTTrait.php:27]: Is there a reason not to use `rest_get_server()`?

```suggestion
		$this->server = rest_get_server();
```

Comment (hellofromtonya) [reply Integration/RESTTrait.php:27]: Hmm, originally I was thinking about bypassing the first filter to ensure we get `WP_Rest_Server`. But .... I think you're right.

Comment (Screenfeed) [reply Integration/RESTTrait.php]: I was only asking actually, I'm not sure we should or shouldn't: both conflict with the precept "store previous state, work, restore stored state" in their current form.

**Pull request merged**